### PR TITLE
Refactor datasqueeze code structure and namespace as well as add publishing to maven central

### PR DESCRIPTION
**Pull request by michaelzhu2003**

### CODE REVIEW

**Comment by yashrajrs** [comment]
Please update the readme to include all the changes. From what I see there are more changes than just publishing to maven central

**Comment by yashrajrs** [comment]
Can we add the DataSqueeze logo commit from our internal code base for readMe as a part of this pull request?

File: pom.xml [hunk 1]
```diff
@@ -74,265 +77,283 @@
...
                 </plugins>
             </build>
         </profile>
     </profiles>
+
+    <repositories>
+        <repository>
+            <id>central</id>
+            <name>public-maven-virtual</name>
+            <url>https://artylab.expedia.biz:443/public-maven-virtual</url>
```
Comment (yashrajrs) [inline pom.xml]: This is internal to Expedia.

**Review by yashrajrs — CHANGES_REQUESTED** [review]

**Comment by yashrajrs** [comment]
The branch fails to build locally

**Comment by michaelzhu2003** [comment]
It builds on my local from both IntelliJ and maven command line.
What is the failure on your local?

File: pom.xml (hunk 1 above)
Comment (michaelzhu2003) [reply pom.xml]: Good catch. Thanks.

File: README.md [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 ## Overview
-DataSqueeze performs compaction of files from source directory to target directory maintaining the directory structure of the source.
+Datasqueeze performs compaction of files from source directory to target directory maintaining the directory structure of the source.
 
 ## Documentation
  This README is intended to provide detailed technical documentation for advanced users.
 
+## Changes since last release
+
+* Edided the pom file for publishing this project's artifacts to the Maven Central
```
Comment (yashrajrs) [inline README.md]: Type - Edited

File: README.md [hunk 3]
```diff
@@ -43,35 +51,35 @@ DataSqueeze supports two types of compaction
...
+## Running Datasqueeze
 
-There are two different ways of running DataSqueeze:
+There are two different ways of running Datasqueeze:
 
 1. CLI -
     a. For TEXT/ORC/SEQ
     ```java
-        hadoop jar dataSqueeze-manager-1.0-SNAPSHOT.jar com.expedia.edw.data.squeeze.Utility
+        hadoop jar Datasqueeze.jar com.expedia.dsp.data.squeeze.Utility
```
Comment (yashrajrs) [inline README.md]: typo - "datasqueeze"

File: README.md [hunk 4]
```diff
@@ -43,35 +51,35 @@ DataSqueeze supports two types of compaction
...
         -sp s3a://edwprod/user/ysontakke/compactiontest1/ -tp s3a://edwprod/user/ysontakke/compactionoutput_text_yash_1/
         -threshold 12345
     ```
 
     b. For AVRO
     ```java
-        hadoop jar dataSqueeze-manager-1.0-SNAPSHOT.jar com.expedia.edw.data.squeeze.Utility
+        hadoop jar Datasqueeze.jar com.expedia.dsp.data.squeeze.Utility
```
Comment (yashrajrs) [inline README.md]: typo - "datasqueeze"

File: README.md [hunk 5]
```diff
@@ -87,22 +95,22 @@ There are two different ways of running DataSqueeze:
        * fileType - Type of file to be compacted (AVRO / TEXT / SEQ / ORC). It is mandatory for AVRO
        * schemaPath - schema used for compaction (mandatory for AVRO)
 
-2. API - [CompactionManager](dataSqueeze-manager/src/main/java/com/expedia/edw/data/squeeze/CompactionManager.java)
+2. API - [CompactionManager](Datasqueeze/src/main/java/com/expedia/dsp/data/squeeze/CompactionManager.java)
```
Comment (yashrajrs) [inline README.md]: typo - "datasqueeze"

File: pom.xml [hunk 6]
```diff
@@ -38,14 +38,17 @@
...
         <apache.commons.io.version>1.3.2</apache.commons.io.version>
         <aws.java.sdk.version>1.11.98</aws.java.sdk.version>
 
         <commons-cli.version>1.2</commons-cli.version>
         <commons.io.version>2.4</commons.io.version>
         <commons.lang3.version>3.0</commons.lang3.version>
 
-        <hadoop.version>2.7.4</hadoop.version>
+        <!-- Use 2.7.4 for github-->
+        <hadoop.version>2.7.1.2.4.3.0-227</hadoop.version>
```
Comment (yashrajrs) [inline pom.xml]: Please revert this change

File: pom.xml [hunk 7]
```diff
@@ -4,31 +4,31 @@
...
+    <artifactId>datasqueeze</artifactId>
+    <name>datasqueeze</name>
     <description>Hadoop utility to compact small files</description>
-    <url>https://expediaincecommerceplatform.github.io/dataSqueeze/</url>
-
-    <packaging>pom</packaging>
-    <version>1.0-SNAPSHOT</version>
+    <url>https://expediaincecommerceplatform.github.io/datasqueeze/</url>
 
+    <version>2.0-SNAPSHOT</version>
```
Comment (yashrajrs) [inline pom.xml]: This needs to move right below artifactId

File: README.md (hunk 3 above)
Comment (michaelzhu2003) [reply README.md]: Great catch. Thanks so much.

File: README.md (hunk 2 above)
Comment (michaelzhu2003) [reply README.md]: Thanks so much.

**Review by baumandm — DISMISSED** [review]

File: README.md [hunk 8]
```diff
@@ -18,19 +27,17 @@ DataSqueeze supports two types of compaction
...
-    Below is a high level summary of the steps that Compaction Utility performs during the course of a typical run for inplace compaction.
+    Below is a high level summary of the steps that Compaction Utility performs during the course of a typical run for in-place compaction.
 
 
         a. Fetch the file paths to be compacted from the source path provided.
         b. Perform mapreduce job using the following configuration
             1. Mapper maps records together based on same parent directory and emits parent directory as key.
-            2. Reducer reduces records based on same key but writes data to the target directory provided by the user, retaining
-                the directory structure.
+            2. Reducer reduces records based on same key but writes data to the target directory provided by the user, retaining the directory structure.
```
Comment (yashrajrs) [inline README.md]: Format looks off

**Comment by yashrajrs** [comment]
Update Pull request description. Looks good otherwise

**Comment by yashrajrs** [comment]
can you also update readMe to say where the jars can be found

**Comment by baumandm** [comment]
> can you also update readMe to say where the jars can be found

This will be easier to update after we actually publish the 2.0 release.  So I would suggest we wait on this.

File: README.md (hunk 8 above)
Comment (baumandm) [reply README.md]: I think the format looks better now than before?

Comment (yashrajrs) [reply README.md]: I would prefer not adding this in the commit range if the text is not updated.
Essentially, keep the master copy for this as is

**Review by baumandm — APPROVED** [review]

**Review by yashrajrs — APPROVED** [review]
LGTM
